neuland/simulation/generator/CosmicMuon.h

```diff
@@ -0,0 +1,137 @@
+        auto rd_num_gen_energy() -> double;
+        auto calculate_abs_momentum(double kinetic_energy) -> double { return kinetic_energy / CLight; };
+        auto calculate_momentum_energy(double kinetic_energy, AngleInfo& angle_info) -> ROOT::Math::PxPyPzE4D<double>;
+
+        auto calculate_external_position_momentum(PointDist point_dist_, AngleDist angle_dist_, EnergyDist energy_dist_)
+            -> std::pair<ROOT::Math::PxPyPzE4D<double>, ROOT::Math::Cartesian3D<double>>;
+
+        auto ReadEvent(FairPrimaryGenerator* prim_gen) -> Bool_t override
+        {
+
+            auto position_momentum = std::pair<ROOT::Math::PxPyPzE4D<double>, ROOT::Math::Cartesian3D<double>>{
+                calculate_external_position_momentum(point_dist_, angle_dist_, energy_dist_)
+            };
+            prim_gen->AddTrack(13,
+                               position_momentum.first.Px(),
+                               position_momentum.first.Py(),
+                               position_momentum.first.Pz(),
+                               position_momentum.second.X(),
+                               position_momentum.second.Y(),
+                               position_momentum.second.Z());
+
+            return true;
+        };
+    };
+
+    template <typename AngleDist, typename EnergyDist, typename PointDist>
+    auto TrackGenerator<AngleDist, EnergyDist, PointDist>::rd_num_gen_angles(AngleDist angle_dist)
+        -> ROOT::Math::Polar3D<double>
+    {
+        auto angles = ROOT::Math::Polar3D<double>{};
+        angles.SetPhi(rd_engine_->Uniform(0., M_PI));
+        angles.SetTheta(angle_dist(rd_engine_));
+        return angles;
+    }
+
+    template <typename AngleDist, typename EnergyDist, typename PointDist>
+    auto TrackGenerator<AngleDist, EnergyDist, PointDist>::calculate_momentum_energy(double kinetic_energy,
+                                                                                     AngleInfo& angle_info)
+        -> ROOT::Math::PxPyPzE4D<double>
+    {
+        auto momentum_energy = ROOT::Math::PxPyPzE4D<double>{ 0, 0, 0, kinetic_energy };
+        auto abs_momentum = double{ calculate_abs_momentum(kinetic_energy) };
+        momentum_energy.SetPx(abs_momentum * angle_info.sin_theta * angle_info.cos_phi);
+        momentum_energy.SetPy(abs_momentum * angle_info.sin_theta * angle_info.sin_phi);
+        momentum_energy.SetPz(abs_momentum * angle_info.cos_theta);
+        return momentum_energy;
+    }
+
+    template <typename AngleDist, typename EnergyDist, typename PointDist>
+    auto TrackGenerator<AngleDist, EnergyDist, PointDist>::calculate_external_position_momentum(PointDist point_dist,
+                                                                                                AngleDist angle_dist,
+                                                                                                EnergyDist energy_dist)
+        -> std::pair<ROOT::Math::PxPyPzE4D<double>, ROOT::Math::Cartesian3D<double>>
```

Comment:
Create a type alias in the PRIVATE section:
```c++
using MomentumPosition = std::pair<ROOT::Math::PxPyPzE4D<double>, ROOT::Math::Cartesian3D<double>>;
```
```suggestion
        -> MomentumPosition
```